### ISSUE TITLE
Collapsible does not share common class structure with HorizontalCollapsible

### ISSUE THREAD

**Issue by czyzby** [kotcrab/vis-editor#253]
These two widgets share similar overall API, but do not implement a common interface or extend an abstract class. Joining these two with a common interface could shorten both of their implementations (no code duplication) and simplify tools like LML.

**Comment by kotcrab** [comment]
I agree. It's kind of messy now because there is `CollapsibleWidget` (since 0.3.1) and `HorizontalCollapsibleWidget` (since 1.2.5). Not sure what to do, I could either:
- break API; rename `CollapsibleWidget` to `VerticalCollapsibleWidget` and add `CollapsibleWidget` interface. If we do renaming we can also consider dropping `Widget` suffix, not sure if it servers any purpose here
- do not rename and add `Collapsible` interface. I would rather avoid `I` prefix for interfaces

What do you think would be better?

**Comment by czyzby** [comment]
There's always `abstract class AbstractCollapsibleWidget`, where you could move the common code. You wouldn't break the API (although this might make sense for naming consistence) and would only slightly affect both of their class hierarchies. This shouldn't be an issue for most users.
